projects/js-packages/charts/src/providers/chart-context/types.ts

```diff
@@ -1,28 +1,34 @@
+import { CSSProperties, ReactNode } from 'react';
 import type { BaseLegendItem } from '../../components/legend';
-import type { CompleteChartTheme } from '../../types';
+import type { CompleteChartTheme, DataPointPercentage, SeriesData } from '../../types';
+import type { LegendShape } from '@visx/legend/lib/types';
+import type { GlyphProps, LineStyles } from '@visx/xychart';
 
 export interface ChartRegistration {
 	legendItems: BaseLegendItem[];
 	chartType: string;
 	metadata?: Record< string, unknown >;
 }
 
+export type GetElementStylesParams = {
+	index: number;
+	data?: SeriesData | DataPointPercentage;
+	overrideColor?: string;
+	legendShape?: LegendShape< SeriesData[], number >;
+};
+
+export type ElementStyles = {
+	color: string;
+	lineStyles: LineStyles;
+	glyph: < Datum extends object >( props: GlyphProps< Datum > ) => ReactNode;
+	shapeStyles: CSSProperties & LineStyles;
+};
+
 export interface GlobalChartsContextValue {
 	charts: Map< string, ChartRegistration >;
 	registerChart: ( id: string, data: ChartRegistration ) => void;
 	unregisterChart: ( id: string ) => void;
 	getChartData: ( id: string ) => ChartRegistration | undefined;
-	/** Theme provided by the GlobalChartsProvider (merged with defaults) */
 	theme: CompleteChartTheme;
-	/**
-	 * Resolve a stable color for a series.
-	 * - If an override color is passed, it wins.
-	 * - If a group is provided, returns a stable color per group across charts.
-	 * - If no group, falls back to index-based color from the theme palette.
-	 */
-	resolveGroupColor: ( params: {
-		group?: string;
-		index: number;
-		overrideColor?: string;
-	} ) => string;
+	getElementStyles: ( params: GetElementStylesParams ) => ElementStyles;
```

Comment:
This is a breaking change that we'll need to account for when upgrading the package.